unknown type name 'uint32_t'

t.h:
```c
#include <stdint.h>
uint32_t test;
```

t.dpp:
```c
#include "t.h"

void main()
{
}
```

```> d++ t.dpp```

output:

```
Error: Error parsing '/tmp/libclangYrt380':
./t.h:3:1: error: unknown type name 'uint32_t'
Program exited with code 1
```

linux + libclang10 + dpp 0.4.9

Including inttypes.h makes things even worse.

--- a/source/dpp/runtime/app.d
+++ b/source/dpp/runtime/app.d
@@ -160,7 +160,7 @@ private TranslationText translationText(in from!"dpp.runtime.options".Options op
         import dpp.expansion: expand, isCppHeader, getHeaderName;
     import clang.util: getTempFileName;
 
-    import std.algorithm: map, filter;
+    import std.algorithm: map, filter, find, startsWith;
     import std.string: fromStringz;
     import std.path: dirName;
     import std.array: array, join;
@@ -181,7 +181,15 @@ private TranslationText translationText(in from!"dpp.runtime.options".Options op
     () @trusted {
         auto includesFile = File(includesFileName, "w");
         foreach(include; includes) {
-            includesFile.writeln(`#include "`, include, `"`);
+            auto paths = includePaths.find!(p => include.startsWith(p));
+            if (paths.length)
+            {
+                includesFile.writeln(`#include <`, include[paths[0].length+1..$], `>`);
+            }
+            else
+            {
+                includesFile.writeln(`#include "`, include, `"`);
+            }
             if(isCppHeader(options, include)) language = Language.Cpp;
         }
     }();

--- a/tests/it/issues.d
+++ b/tests/it/issues.d
@@ -1934,3 +1934,53 @@ version(Linux) {
     );
 
 }
+
+@Tags("issue")
+@("307")
+@safe unittest {
+    with(immutable IncludeSandbox()) {
+        import std.process : environment;
+        import std.string : join;
+
+        writeFile("includes/dir1/issue307.h",
+                  `
+                    #ifndef ISSUE307_H
+                        #define ISSUE307_H
+                        #if __has_include_next(<issue307.h>)
+                            #include_next <issue307.h>
+                        #else // fallback for cpp that does not support include_next
+                            typedef int myint_t;
+                        #endif
+                    #endif
+                  `);
+        writeFile("includes/dir2/issue307.h",
+                  `
+                    typedef int myint_t;
+                  `);
+        writeFile("issue307.dpp",
+                  `
+                   #include <issue307.h>
+                  `);
+
+        auto original_cpath = environment.get("CPATH", "");
+        environment["CPATH"] = [
+            inSandboxPath("includes/dir1"),
+            inSandboxPath("includes/dir2"),
+            original_cpath,
+        ].join(":");
+        scope(exit)
+        {
+            if (original_cpath == "")
+            {
+                environment.remove("CPATH");
+            }
+            else
+            {
+                environment["CPATH"] = original_cpath;
+            }
+        }
+
+        runPreprocessOnly("issue307.dpp");
+        fileShouldContain("issue307.d", q{alias myint_t = int;});
+    }
+}

--- a/tests/it/issues.d
+++ b/tests/it/issues.d
@@ -1939,6 +1939,7 @@ version(Linux) {
 @("307")
 @safe unittest {
     with(immutable IncludeSandbox()) {
+        import std.path : pathSeparator;
         import std.process : environment;
         import std.string : join;
 
@@ -1967,7 +1968,7 @@ version(Linux) {
             inSandboxPath("includes/dir1"),
             inSandboxPath("includes/dir2"),
             original_cpath,
-        ].join(":");
+        ].join(pathSeparator);
         scope(exit)
         {
             if (original_cpath == "")

